src/main/resources/static/assets/js/sdh.js

```diff
@@ -267,18 +250,14 @@ class ModalEditRow {
 		    	// Persistence is not currently required. Just update the display.
 		    	row.item.flagged = !row.item.flagged;
 		    },
-		    edit(item, index, target) {
-		    	const domain = this._selectValue(this.domainOptions, item.domain);
-	    		const referred = this._selectValue(this.yesNoOptions, item.referred);
-	    		const referralComplete = this._selectValue(this.yesNoOptions, item.referral_complete);
-		    	if (!item.id) {
-		    		this.modalEditRow.setFields('New Clinician Review', null, domain, '', '', referred, referralComplete);
+		    edit(index) {
+		    	// Don't allow the user to click away from a row until they've saved.
+		    	if (this.rowToEdit != null && this.rowToEdit !== index) {
+		    		alert('Click the save icon to save your changes.')
```

Comment:
Possibly too much work for this project, but a nicer interaction would be to allow the user to choose whether to save or abandon their changes (using `confirm` rather than `alert`).

Reply:
The way the model is defined, the fields were changing on screen when they clicked to edit another row making it appear as if the info was saved. Rather than try to unwind it, I took the shortcut of forcing them to save. But agreed it's not ideal.